### PR TITLE
Datadog StatsD Reporter

### PR DESCRIPTION
**Pull request by cozos**
Still need to write tests, but what do you think about this?

Also:
- Need a way to determine if we want to use gauge/timing/increment/etc, [statsd-jvm-profiler](https://github.com/etsy/statsd-jvm-profiler/blob/master/src/main/java/com/etsy/statsd/profiler/reporter/StatsDReporter.java#L45) seems to only care about gauge values, so maybe that's all thats needed.
- Need a way to get tags (i.e. tag by YARN container, application)

### CODE REVIEW

**Comment by felixcheung** [comment]
could you submit the CLA? thanks for the contribution

**Comment by cozos** [comment]
Submitted the CLA. I'll write some tests soon.

File: src/main/java_datadog/com/uber/profiling/reporters/DatadogOutputReporter.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        }
+    }
+
+    // properties from yaml file
+    @Override
+    public void updateArguments(Map<String, List<String>> connectionProperties) {
+        for (Map.Entry<String,  List<String>> entry : connectionProperties.entrySet()) {
+            String key = entry.getKey();
+            List<String> value = entry.getValue();
+            if (StringUtils.isNotEmpty(key) && value != null && !value.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (cozos) [reply src/main/java_datadog/com/uber/profiling/reporters/DatadogOutputReporter.java]: `value` is a `List`, not a String. I'm not quite sure why, the interface for `updateArguments` is like that and I just followed the same pattern as the `InfluxDB` reporter. 

I added a `StringUtils.isBlank` check on the `stringValue` though: https://github.com/uber-common/jvm-profiler/pull/40/commits/66b12ccc755eca432a4ae0aae621c2928d4cff9f

**Comment by boy-uber** [comment]
Great, thanks Arwin for the contribution! Let me merge your changes.

**Pull request merged**